File: find_recent.js
const fs = require('fs');
const path = require('path');
const os = require('os');

const searchPaths = [
    path.join(os.homedir(), 'Desktop'),
    path.join(os.homedir(), 'Downloads'),
    path.join(os.homedir(), 'Pictures')
];

let recentFiles = [];
const now = Date.now();

searchPaths.forEach(searchPath => {
    if (fs.existsSync(searchPath)) {
        const files = fs.readdirSync(searchPath);
        files.forEach(file => {
            const filePath = path.join(searchPath, file);
            try {
                const stats = fs.statSync(filePath);
                if (stats.isFile() && (now - stats.mtimeMs < 600000)) { // 10 mins
                    if (['.png', '.jpg', '.jpeg', '.webp'].includes(path.extname(file).toLowerCase())) {
                        recentFiles.push({ time: stats.mtimeMs, path: filePath });
                    }
                }
            } catch (e) {}
        });
    }
});

recentFiles.sort((a, b) => b.time - a.time);
recentFiles.slice(0, 5).forEach(f => {
    console.log(`${new Date(f.time).toLocaleTimeString()}: ${f.path}`);
});

File: find_recent.py
import os
import time

current_time = time.time()
search_paths = [
    os.path.expanduser("~/Desktop"),
    os.path.expanduser("~/Downloads"),
    os.path.expanduser("~/Pictures")
]

recent_files = []

for path in search_paths:
    if os.path.exists(path):
        for root, dirs, files in os.walk(path):
            for file in files:
                file_path = os.path.join(root, file)
                try:
                    mtime = os.path.getmtime(file_path)
                    if current_time - mtime < 600: # Last 10 minutes
                        if file.lower().endswith(('.png', '.jpg', '.jpeg', '.webp')):
                            recent_files.append((mtime, file_path))
                except:
                    continue

recent_files.sort(reverse=True)
for mtime, path in recent_files[:10]:
    print(f"{time.ctime(mtime)}: {path}")
